VideoLocker/src/main/java/org/edx/mobile/view/adapters/DiscussionCommentsAdapter.java

```diff
@@ -107,7 +144,34 @@ public void run() {
 
     @Override
     public int getItemCount() {
```

Comment:
This function confuses me a bit. What is `progressVisible`, does that mean the loading indicator? And for this particular function, what is it getting the count for?

A lot of this counting logic seems pretty confusing. Are there any ways to add unit tests for the counting?